dockerv2.go

```diff
@@ -0,0 +1,49 @@
+package txtdirect
+
+import (
+	"fmt"
+	"net/http"
+	"regexp"
+	"strings"
+)
+
+var dockerRegexs = map[string]string{
+	"v2":        "^\\/?v2\\/?$",
+	"_catalog":  "^/v2/_catalog$",
+	"tags":      "^/v2/(.*)/tags/(.*)",
+	"manifests": "^/v2/(.*)/manifests/(.*)",
+	"blobs":     "^/v2/(.*)/blobs/(.*)",
+}
+
+var DockerRegex = regexp.MustCompile("^\\/v2\\/(.*\\/(tags|manifests|blobs)\\/.*|_catalog$)")
+
+func redirectDockerv2(w http.ResponseWriter, r *http.Request, rec record) error {
+	path := r.URL.Path
+	v2Regex, err := regexp.Compile(dockerRegexs["v2"])
```

Comment:
This could be precompiled as well, instead of compiling it everytime we call the function ?

I haven't tested this, but could it be possible to have something like this for `dockerRegexs` ?

```go
var dockerRegexs = map[string]Regexp{
	"v2":        regex.MustCompile("^\\/?v2\\/?$"),
	"_catalog":  regex.MustCompile("^/v2/_catalog$"),
	"tags":      regex.MustCompile("^/v2/(.*)/tags/(.*)"),
	"manifests": regex.MustCompile("^/v2/(.*)/manifests/(.*)"),
	"blobs":     regex.MustCompile("^/v2/(.*)/blobs/(.*)"),
}
```

The map *should* be able to contain any type we throw at it, I'm just not sure we can initialize it this way. (if we can't, maybe in an `init()` function ?